test_unstructured_ingest/unit/pipeline/reformat/test_chunking.py

```diff
@@ -0,0 +1,144 @@
+from __future__ import annotations
+
+import os
+import tempfile
+
+import pytest
+from _pytest.logging import LogCaptureFixture
+
+from test_unstructured.unit_utils import (
+    FixtureRequest,
+    Mock,
+    example_doc_path,
+    function_mock,
+    method_mock,
+)
+from unstructured.documents.elements import CompositeElement
+from unstructured.ingest.interfaces import ChunkingConfig, PartitionConfig
+from unstructured.ingest.pipeline.interfaces import PipelineContext
+from unstructured.ingest.pipeline.reformat.chunking import Chunker
+
+ELEMENTS_JSON_FILE = example_doc_path(
+    "test_evaluate_files/unstructured_output/Bank Good Credit Loan.pptx.json"
+)
+
+
+class DescribeChunker:
+    """Unit tests for ingest.pipeline.reformat.chunking.Chunker"""
+
+    # -- Chunker.run() -----------------------------------------------------------------------------
+
+    # -- integration test --
+    def it_creates_json(self, _ingest_docs_map_: Mock):
+        chunking_config = ChunkingConfig(chunking_strategy="by_title")
+        pipeline_context = PipelineContext()
+        partition_config = PartitionConfig()
+        chunker = Chunker(
+            chunking_config=chunking_config,
+            pipeline_context=pipeline_context,
+            partition_config=partition_config,
+        )
+        with tempfile.TemporaryDirectory() as tmpdir:
```

Comment:
Hmm, no, I don't think this is how you want to do the temporary directory thing.

`tmpdir` is a "built-in" `pytest` fixture. So if you include `tmpdir` in the parameter list for `it_creates_json()` then `pytest` will do all the needful for you. The rest of this can be the same, you just ditch the `with tempfile.TemporaryDirectory()` bit. If you use that you need to do everything in the "with" context and there's no way to inspect the results afterward. Also it's more work.

If you look into it in the `pytest` documentation you'll see `pytest` does something like store the temp-dirs for the last five test runs. So it never gets out of control but you can always inspect the results if you want.